front/app/components/IdeaButton/index.tsx

```diff
@@ -323,16 +323,22 @@ const IdeaButton = memo<Props & InjectedIntlProps>(
                   disabled={!enabled}
                   ariaDisabled={false}
                 >
-                  <FormattedMessage
-                    {...getInputTermMessage(inputTerm, {
-                      idea: messages.submitYourIdea,
-                      option: messages.addAnOption,
-                      project: messages.addAProject,
-                      question: messages.addAQuestion,
-                      issue: messages.submitAnIssue,
-                      contribution: messages.addAContribution,
-                    })}
-                  />
+                  {project.attributes.participation_method === 'ideation' && (
+                    <FormattedMessage
+                      {...getInputTermMessage(inputTerm, {
+                        idea: messages.submitYourIdea,
+                        option: messages.addAnOption,
+                        project: messages.addAProject,
+                        question: messages.addAQuestion,
+                        issue: messages.submitAnIssue,
+                        contribution: messages.addAContribution,
+                      })}
+                    />
+                  )}
+                  {project.attributes.participation_method ===
+                    'native_survey' && (
+                    <FormattedMessage {...messages.takeTheSurvey} />
+                  )}
```

Comment:
It seems like we'll need to do a "_if native_survey ..._" check to control the messages at some point. Unless there is another cleaner way to do this?

Reply:
Delegation to a new abstraction `ParticipationMethod`, like we did in the back-end (but not in the native survey branch yet) is a way to avoid if statements like this. I do not know how to mix that with `tsx` files yet, but we should talk about it more to come up with a good solution.

Please note that phases have a participation_method` too, not only projects. Nothing to worry about that for iteration 1, but keep it in mind. Maybe in iteration 2 we will already work on surveys in phases.